lib/sla.js

```diff
@@ -98,75 +99,156 @@ function addChannelsToBridge(inbound, dialed, bridge) {
  * @param {Object} bridge - Object that defines the bridge to be passed to
  *   addChannelsToBridge().
+ * @ param {Object} trunkData - the data of the trunk to be used for station
+ *   access.
  * @return {Q} Q promise object.
  */
-function originateChannel(client, inbound, bridge) {
+function originateChannel(client, inbound, bridge, trunkData) {
   var defer = Q.defer();
-  var dialed = [client.Channel(), client.Channel()];
-  var originate = [];
-  var numToHangup = dialed.length;
-  var station = ['SIP/phone1', 'SIP/phone2'];
-  dialed.forEach(function (channel, index, dialed) {
-    originate.push(Q.denodeify(dialed[index].originate.bind(dialed[index])));
-    originate[index]({endpoint: station[index], app: 'sla', appArgs: 'dialed',
-        timeout: 30})
-      .catch(function (err) {
-        err.name = 'OriginateFailure';
-        defer.reject(err);
-      });
-    dialed[index].once('StasisStart', function(event, line) {
-      var answer = Q.denodeify(line.answer.bind(line));
-      answer();
-      var toKill = dialed.filter(function (unanswered) {
-        return unanswered.id !== line.id;
-      });
-      toKill.forEach(function (unanswered) {
-        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
-        hangup();
-      });
-      defer.resolve(line);
+  var dialed = [];
+  /**
+   * The function that gets called once dialed enters Stasis.
+   *   Kills the other remaining stations and returns the line that picked up.
+   * @param {Object} event - the information related to the StasisStart event
+   * @param {Object} line - the line that has entered Stasis
+   */
+  function dialedEnteredStasis(event, line) {
+    var answer = Q.denodeify(line.answer.bind(line));
+    answer();
+    var toKill = dialed.filter(function (unanswered) {
+      return unanswered.channel.id !== line.id;
     });
-    inbound.once('ChannelHangupRequest', function(event, line) {
-      defer.resolve('Inbound channel hungup');
-      dialed.forEach(function (unanswered) {
-        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
-        hangup();
-      });
+    toKill.forEach(function (unanswered) {
+      unanswered.channel.removeListener('ChannelDestroyed',
+        dialedChannelHungup);
+      var hangup = Q.denodeify(unanswered.channel.hangup.bind(
+          unanswered.channel));
+      hangup();
     });
-    dialed[index].once('ChannelDestroyed', function(event) {
-      originate.pop(dialed[index]);
-      if(originate.length === 0) {
-        defer.resolve('All dialed channels hungup');
-        var hangup = Q.denodeify(inbound.hangup.bind(inbound));
-        hangup();
+    defer.resolve(line);
+  }
+  /**
+   * Utility function for cheking if all dialed channels have been hungup
+   * @param {Array} dialed - the array of channels; the length is checked
+   * @return {boolean} - whether or not the dialed array is empty
+   */
+
+  var allAreHungup = function(dialed) {
+   if (dialed.length) {
```

Comment:
You can just use:

return dialed.length;

to get the same result. Although I don't think it's doing what you think it's doing. Any integer above 0 will evaluate to true, so if the array has channels that have not been hungup yet, it will evaluate that all channels have been hungup. You should do !dialed.length instead so it only evaluates to true once the array has had all its elements removed.